cli/crates/cli/src/main.rs

```diff
@@ -185,5 +186,21 @@ fn try_main(args: Args) -> Result<(), CliError> {
         SubCommand::DumpConfig => dump_config::dump_config(),
         SubCommand::Check(cmd) => check::check(cmd),
         SubCommand::Trust(cmd) => trust::trust(cmd),
+        SubCommand::Upgrade => {
+            // this command is also hidden in this case
+            // (clippy doesn't have a mechanism to completely disable a command conditionally when using derive, see https://github.com/clap-rs/clap/issues/5251)
+            if is_direct_install() {
+                return Err(CliError::NotDirectInstall);
+            }
+            upgrade::install_grafbase().map_err(Into::into)
+        }
     }
 }
+
+pub(crate) fn is_direct_install() -> bool {
+    std::env::current_exe().is_ok_and(|path| Some(path) != direct_install_executable_path())
```

Comment:
It may be me not grokking the logic, but from the function name, I would expect `Some(path) == ...`.

Reply:
You're right, I meant to rename the function to `is_not_direct_install` since I'm not sure you can negate a `hide` condition

Reply:
Done